transport/http/inbound.go

```diff
@@ -94,9 +100,15 @@ func (i *inbound) Stop() error {
 	if i.listener == nil {
 		return nil
 	}
+	if !atomic.CompareAndSwapUint32(&i.stopping, 0, 1) {
+		return nil
+	}
 	err := i.listener.Close()
 	i.listener = nil
-	<-i.done
+	if serveErr := <-i.done; err == nil && serveErr != nil {
+		err = serveErr
+	}
+	atomic.CompareAndSwapUint32(&i.stopping, 1, 0)
```

Comment:
I'm not sure what the advantage in flipping this is? Once we stop, we don't need to stop again, so I'd prefer to rename the variable to "stopped" and only execute the stop logic if it was 0 previously